src/frontend/src/components/RecordSearch/index.tsx

```diff
@@ -2,14 +2,103 @@ import React from 'react';
 import { connect } from "react-redux";
 import { AppState } from "../../redux/store";
 import { loadRecords } from "../../redux/records/actions";
-
-// This is a placeholder component. I created it to prototype
-// out the general structure we'll need when working with
-// Records using Redux.
+import moment from 'moment';
 
 class RecordSearch extends React.Component {
+  state = {
+    firstName: null,
+    lastName: null,
+    dateOfBirth: '', // Moment expects a string to be passed in as a paramenter in the validateForm function.
+    missingFirstName: undefined, // Undefined is used here to prevent aria-invalid from rendering until form submission.
+    missingLastName: undefined,
+    missingInputs: null,
+    invalidDate: undefined
+  }
+
+  handleChange = (e: React.BaseSyntheticEvent) => {
+    this.setState({
+      [e.target.id]: e.target.value
+    });
+  }
+
+  handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    this.validateForm();
+    if (this.state.missingInputs === false && this.state.invalidDate === false) {
+      // Dispatch an action.
+    };
+  }
+
+  validateForm = () => {
+    !this.state.firstName ? this.setState({missingFirstName: true}) : this.setState({missingFirstName: false});
+    !this.state.lastName ? this.setState({missingLastName: true}) : this.setState({missingLastName: false});
```

Comment:
`!' '` is `false`. Is this intended behavior for when there is only whitespace in the form field?

Reply:
Well, the behavior was intended but my variable naming was unclear.  
`missingFirstName` -> `firstNameHasInput`
So, an empty form field  should set `firstNameHasInput: false`

Reply:
That's fine and you don't have to change the names! Just wanted to make sure since otherwise the validations would consider whitespace as valid, wasn't sure if that was intended behavior.

Reply:
Ahhhh, I see what you mean now.  This was not intended. I'll try to account for this case.